lib/api/api_auth_webauthn.go

```diff
@@ -468,7 +459,12 @@ func (s *webauthnService) loadVolatileState() *WebauthnVolatileState {
 	return &state
 }
 
-func (s *webauthnService) storeVolatileState(state *WebauthnVolatileState) error {
+func (s *webauthnService) updateVolatileState(update func(state *WebauthnVolatileState)) error {
+	s.volStateMut.Lock()
+	defer s.volStateMut.Unlock()
```

Comment:
Do we need to keep the lock held until the DB operation has completed?

Reply:
Yes, otherwise we might get a concurrent write conflict, right? Or am I misunderstanding the question?

Reply:
Nah, probably fine.  It's done similarly in other packages.